data_steward/retraction/retract_deactivated_pids.py

```diff
@@ -169,430 +143,221 @@ def get_table_dates_info(project_id, dataset_id, client):
-
-    :param project_id: Identifies the project where data is being retracted
-    :param ticket_number: Jira ticket number to identify and title sandbox table
-    :param pids_project_id: Identifies the project containing deactivated pids table
-    :param pids_dataset_id: Identifies the dataset containing deactivated pids table
-    :param pids_table: Name of the deactivated pids table. This table should have
-      the following fields: (person_id: int, suspension_status: string, deactivated_date: date).
-    :param datasets: (optional) List of datasets to retract from. If not provided,
-      retraction will be performed from all datasets in project referred to by `project_id`.
-    :return: list of query dictionaries
-
-    NOTE: For dataset_ids matching `retraction.retract_utils.DEID_REGEX`, associated research_ids
-    retrieved from an inferred combined dataset are used for retraction.
+    Creates queries for sandboxing and deleting records
+
+    :param client: BigQuery client
+    :param project_id: Identifies the project
+    :param dataset_id: Identifies the dataset to retract deactivated participants from
+    :param sandbox_dataset_id: Identifies the dataset to store records to delete
+    :param deact_pids_table_ref: BigQuery table reference to dataset containing deactivated participants
+    :param pid_rid_table_ref: BigQuery table reference to dataset containing pid-rid mappings
+    :return: List of query dicts
+    :raises:
+        RuntimeError: 1. If retracting from deid dataset, pid_rid table must be specified
+                      2. If mapping or ext table does not exist, EHR data cannot be identified
     """
-    queries_list = []
-    dataset_list = set()
-    final_date_column_df = pd.DataFrame()
-    # Hit bq and receive df of deactivated ehr pids and deactivated date
-    client = bq.get_client(project_id)
-    deactivated_ehr_pids_df = client.query(
-        DEACTIVATED_PIDS_QUERY.render(project=pids_project_id,
-                                      dataset=pids_dataset_id,
-                                      table=pids_table)).to_dataframe()
-    if datasets is None:
-        date_columns_df = get_date_info_for_pids_tables(project_id, client)
-    else:
-        date_columns_df = get_date_info_for_pids_tables(project_id, client,
-                                                        datasets)
-    LOGGER.info(
-        "Dataframe creation complete. DF to be used for creation of retraction queries."
-    )
-    for date_row in date_columns_df.itertuples(index=False):
-        # Filter to only include tables containing deactivated pids with the earliest deactivated date
-        LOGGER.info(
-            f'Checking table: {date_row.project_id}.{date_row.dataset_id}.{date_row.table}'
+    table_cols_df = get_table_cols_df(client, project_id, dataset_id)
+    table_dates_info = get_table_dates_info(table_cols_df)
+    tables = table_cols_df['table_name'].to_list()
+    is_deid = ru.is_deid_dataset(dataset_id)
```

Comment:
With this test command line, this raises an error that it shouldn't raise.
```
PYTHONPATH=./:$PYTHONPATH python retraction/retract_deactivated_pids.py -p aou-res-curation-test -d lrwb_deid_test_input_20190910 -a aou-res-curation-test.lrwb_deid_test_input_20190910._deactivated_participants
```
This raises the `RuntimeError` even though the dataset is not a deid dataset.  This may be something to bring up at an eng sync, but I would prefer to base the dataset type on labels instead of relying on names.  What do you think?

Reply:
I would be happy to use labels instead of names. My only hesitation is that this could potentially be run on past datasets, which may have no label information. If we have already added labels to our past datasets, I can make this change.

Reply:
After discussion, labels must be prioritized and names should be a fallback.